IoTHubBuddy/IoTViewer/MainPage.xaml.cs

```diff
@@ -0,0 +1,131 @@
+//NOTE: Authentication doesn't really work right now, so if you don't have a corporate/organization azure account
+//please fill in the necessary strings and uncomment NavigateToMap() in On_Loaded()
+
+namespace IoTHubBuddy
+{
+    /// <summary>
+    /// An empty page that can be used on its own or navigated to within a Frame.
+    /// </summary>
+    public sealed partial class MainPage : Page
+    {
+        
+        
+        public MainPage()
+        {
+            this.InitializeComponent();
+            string URI = string.Format("ms-appx-web://Microsoft.AAD.BrokerPlugIn/{0}", WebAuthenticationBroker.GetCurrentApplicationCallbackUri().Host.ToUpper());
+            
+
+        }
+        /// <summary>
+        /// event handler for login in click, will bring up login window
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void LoginButton_Click(object sender, RoutedEventArgs e)
+        {
+            AccountsSettingsPane.Show();
+        }
+        
+        protected override void OnNavigatedTo(NavigationEventArgs e)
+        {
+            AccountsSettingsPane.GetForCurrentView().AccountCommandsRequested += BuildPaneAsync;
+
+        }
+        protected override void OnNavigatedFrom(NavigationEventArgs e)
+        {
+            AccountsSettingsPane.GetForCurrentView().AccountCommandsRequested -= BuildPaneAsync;
+        }
+        /// <summary>
+        /// populate the account window pane with cached accounts and the add account button
+        /// </summary>
+        /// <param name="s"></param>
+        /// <param name="e"></param>
+        private async void BuildPaneAsync(AccountsSettingsPane s,
+    AccountsSettingsPaneCommandsRequestedEventArgs e)
+        {
+            AccountManager.OnAccountCommandsRequested(s, e, this);
+            
+        }
+        /// <summary>
+        /// load an instance of MapPage.xaml
+        /// </summary>
+        /// <param name="name"></param>
+        public void NavigateToMap(IoTAccountData test)
+        {
+            this.Frame.Navigate(typeof(MapPage), test);
+        }
+        /// <summary>
+        /// load an instance of Subscription.xaml
+        /// </summary>
+        /// <param name="token"></param>
+        public void NavigateToSubscription(string token)
+        {
+            this.Frame.Navigate(typeof(SubscriptionPage), token);
+        }
+        /// <summary>
+        /// sign in silently first. If that fails, then bring up gui
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private async void On_Loaded(object sender, RoutedEventArgs e)
+        {
+            List<string> partitionIds = new List<string>();
+            partitionIds.Add("0");
+            partitionIds.Add("1");
+            string entity = "iothub-ehub-hub-dimaha-42362-31f5229b77"; //should start with "iothub-ehub-<hub name>-<random numbers>"
```

Comment:
remember to not check this in. 
any customizable code should be in one place, so it's easy to find.